updater/spec/dependabot/dependency_change_builder_spec.rb

```diff
@@ -105,9 +105,9 @@
       end
     end
 
-    context "when the source is a group rule" do
+    context "when the source is a dependency group" do
       let(:change_source) do
-        Dependabot::GroupRule.new(name: "dummy-pkg-*")
+        Dependabot::DependencyGroup.new(name: "dummy-pkg-*", rules: ["dummy-pkg-*"])
```

Comment:
I've had to refer back to the API, and I believe the actual data structure we get here would be more like:

```suggestion
        Dependabot::DependencyGroup.new(name: "dummy-pkg-*", rules: { "patterns" => "dummy-pkg-*"]})
```

i.e the configuration file would have the form:

```yaml
version: 2
updates:
  - package-ecosystem: "bundler"
    directory: "/"
    schedule:
      interval: "daily"
    groups:
      - dummy-pkg-*:
          patterns:
            - "dummy-pkg-*"
```

We landed on this after a few passed in order to allow for rules which used things other than pattern-based matching as we iterate.

For purposes of this iteration in Core/Updater, I think it is fine to keep our code-as is and just strip out the value of `group["rules]["patterns"]` as the only thing we care about right now.

Reply:
Thanks I actually got that [wrong](https://github.com/dependabot/cli/pull/94) in the CLI too. I'll propose a fix there and make an adjustment here

Reply:
Adjusted in 9ab1df5a11c9a74d48d22e5b73eeb9df1c002f2f

Reply:
Fixed for the CLI in https://github.com/dependabot/cli/pull/95